src/coreclr/src/vm/genericdict.cpp

```diff
@@ -809,10 +780,10 @@ Dictionary* Dictionary::GetMethodDictionaryWithSizeCheck(MethodDesc* pMD, ULONG
     }
     CONTRACT_END;
 
-    Dictionary* pDictionary = pMD->GetMethodDictionary();
```

Comment:
This is where the race condition was. We have fetched the Dictionary, then some other thread expanded the dictionary, and then we fetched the number of slot from the new Dictionary; but still kept using the old Dictionary.

Reply:
Thanks Jan for tracking this down!